simpleviewinc/tap-events-force
# filter results counter

**Ticket**: [issue_id](https://jira.simpleviewtools.com/browse/ZEN-375)

`[semver]: minor`

## Context

* Filter Modal specs has changed since we last sized them
* it should also include a 'results' counter (see attached image)
* it should be styled with the same style as the Group Booking Modal - "Places Remaining" counter

## Goal

* add a 'results' counter in the location as seen in the attached image
* styling should be the same as the counter on group booking modal
* Filtering logic should execute before 'APPLY' is pressed because we need to be able to get the count
* get the count number by filtering on the sessions item state
* if results is 0. the 'APPLY' button should be disabled

## Updates

- `src/components/modals/filter.js`
   * updated apply button to be disabled when results == 0
   * updated the classname of the TopSection to be the correct ones
   * added results counter to the top section
   * performs the filtering after each label select/deselect. but does not apply them to the sessionAgenda until Apply is hit

- `src/theme/components/modals/filterModal.js`
   * styling

- added some unit tests to the filtering utility function
 
## Testing

* run `keg dpg run package=docker.pkg.github.com/simpleviewinc/keg-packages/tap:zen-375-results-counter`
* open the filter modal
* start selecting the labels
* verify that you see the results counter appear with a number
    * ![image](https://user-images.githubusercontent.com/3317835/96913768-efee9500-1458-11eb-8080-567f7553612c.png)

* Apply it
* verify that the amount of sessions displayed matches the count number from the filter modal

### Result 0 test
* open the filter modal
* select a label combination that leads to result = 0
* Verify that the `Apply` button is disabled

### Unit Test
* run `keg dpg run package=docker.pkg.github.com/simpleviewinc/keg-packages/tap:zen-375-results-counter command=test`
* verify all test passes
